core/src/main/scala/org/apache/spark/rdd/RDD.scala

```diff
@@ -140,7 +140,7 @@ abstract class RDD[T: ClassTag](
   val id: Int = sc.newRddId()
 
   /** A friendly name for this RDD */
-  @transient var name: String = null
+  @transient var name: String = Utils.getFormattedClassName(this)
```

Comment:
Hmm - will this change existing behavior? It looks like in `toString` of RDD we check whether this is null. Will this change the string form of an RDD:

```
  override def toString: String = "%s%s[%d] at %s".format(
    Option(name).map(_ + " ").getOrElse(""), getClass.getSimpleName, id, getCreationSite)
```